src/nci_adapter.c

```diff
@@ -88,11 +90,20 @@ G_DEFINE_ABSTRACT_TYPE(NciAdapter, nci_adapter, NFC_TYPE_ADAPTER)
 
 #define PRESENCE_CHECK_PERIOD_MS (250)
 
+#define FIELD_SIZEOF(t, f) (sizeof(((t*)0)->f))
+#define RANDOM_UID_SIZE (4)
+#define RANDOM_UID_START_BYTE (0x08)
+
 /*==========================================================================*
  * Implementation
  *==========================================================================*/
 
-#define nci_adapter_intf_info_free(x) g_free(x)
+#define nci_adapter_intf_info_free(x)               \
+do {                                                \
+    if (x && x->mode_param_parsed)                  \
+        g_free(x->mode_param_parsed);               \
+    g_free(x);                                      \
+} while (0);
```

Comment:
Let's kill this macro and just do
```c
            g_free(priv->active_intf->mode_param_parsed);
            g_free(priv->active_intf);
```
in the only place where it's actually used.

Reply:
done

Reply:
But the macro is still alive!

Reply:
yeap, but now it is dead :)